extensions/parquet/table/src/main/java/io/deephaven/parquet/table/util/ByteBufferAsyncResponseTransformer.java

```diff
@@ -0,0 +1,89 @@
+package io.deephaven.parquet.table.util;
+
+import io.deephaven.base.verify.Assert;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import software.amazon.awssdk.core.async.AsyncResponseTransformer;
+import software.amazon.awssdk.core.async.SdkPublisher;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * An {@link AsyncResponseTransformer} that transforms a response into a {@link ByteBuffer}.
+ * This class is inspired from {@link software.amazon.awssdk.core.internal.async.ByteArrayAsyncResponseTransformer}
+ * but avoids a number of extra copies done by the former.
+ *
+ * @param <ResponseT> POJO response type.
+ */
+public final class ByteBufferAsyncResponseTransformer<ResponseT> implements AsyncResponseTransformer<ResponseT, ByteBuffer> {
+
+    private volatile CompletableFuture<ByteBuffer> cf;
+    private final ByteBuffer byteBuffer;
+
+    ByteBufferAsyncResponseTransformer(final int bufferSize) {
+        // TODO Can be improved with a buffer pool
+        byteBuffer = ByteBuffer.allocate(bufferSize);
+    }
+
+    @Override
+    public CompletableFuture<ByteBuffer> prepare() {
+        cf = new CompletableFuture<>();
+        return cf;
+    }
```

Comment:
I want to avoid reallocations, do you think it would be okay if I clear the buffer in each prepare call?
Something like:
```
    @Override
    public CompletableFuture<ByteBuffer> prepare() {
        byteBuffer.clear();
        cf = new CompletableFuture<>();
        return cf;
    }
```

Reply:
It's unclear to me if this is safe without digging into the semantics of the retry mechanisms. If we do start using buffer pools, we'd probably need to dig in to figure out when it's safe to release the buffer back to the pool in retry cases. I don't think it's worth the risk of re-using the buffer at the moment, and I don't think I'd sweat the extra allocation since I'm guessing the majority of requests succeed without retry.

Reply:
I have updated the design slightly to make it same as `software.amazon.awssdk.core.internal.async.ByteArrayAsyncResponseTransformer`, which is very similar to your suggestion. We will create a new `CompletableFuture` on `prepare()` and a new `ByteBufferSubscriber` on `onStream`.  Let me know if you think this can work.